static/app/views/releases/thresholdsList/thresholdGroupRows.tsx

```diff
@@ -213,56 +206,45 @@ export function ThresholdGroupRows({
   return (
     <StyledThresholdGroup>
       {Array.from(thresholdIdSet).map((tId: string, idx: number) => {
-        const threshold = editingThresholds[tId] || thresholdsById[tId];
-        let environmentName;
-        if (editingThresholds[tId]) {
-          // editing environment type is String
-          environmentName = editingThresholds[tId].environment;
+        let threshold;
+        let isEditing = false;
+        if (tId in editingThresholds) {
+          isEditing = true;
+          threshold = editingThresholds[tId] as EditingThreshold;
         } else {
-          // Threshold environment type is Environnment
-          environmentName = thresholdsById[tId].environment
-            ? thresholdsById[tId].environment.name
-            : '';
+          threshold = initialThreshold as Threshold;
         }
+
         return (
           <StyledRow
             key={threshold.id}
-            lastRow={idx === thresholdIdSet.size - 1}
-            hasError={threshold.hasError}
+            lastRow={isLastRow && idx === thresholdIdSet.size - 1}
+            hasError={isEditing && (threshold as EditingThreshold).hasError}
           >
-            <FlexCenter style={{borderBottom: 0}}>
-              {idx === 0 ? (
-                <ProjectBadge
-                  project={threshold.project}
-                  avatarSize={16}
-                  hideOverflow
-                  disableLink
-                />
-              ) : (
-                ''
-              )}
-            </FlexCenter>
-            {/* TODO: IF this is a new threshold - allow environment select */}
-            {newGroup ? (
+            {/* ENV ONLY EDITABLE IF NEW */}
+            {!initialThreshold || threshold.id !== initialThreshold.id ? (
               <CompactSelect
                 style={{width: '100%'}}
-                value={threshold.environment}
+                value={threshold.environmentName}
                 onChange={selectedOption =>
                   editThresholdState(threshold.id, 'environment', selectedOption.value)
                 }
-                options={newGroup.environments.map(env => ({
+                options={allEnvironmentNames.map(env => ({
                   value: env,
                   textValue: env,
                   label: env,
                 }))}
               />
             ) : (
-              <FlexCenter style={{borderBottom: 0}}>
-                {idx === 0 ? environmentName || 'None' : ''}
+              <FlexCenter>
+                {/* 'None' means it _has_ an environment, but the env has no name */}
+                {threshold.environment
+                  ? threshold.environment.name || 'None'
+                  : '{No environment}'}
               </FlexCenter>
             )}
             {/* FOLLOWING COLUMNS ARE EDITABLE */}
-            {editingThresholds[threshold.id] ? (
+            {isEditing ? (
```

Comment:
Should isEditing be a state variable? In other words, will this component stay either editing or none editing for one single threshold? or can is it expected to switch in between?

Reply:
yea - this is per threshold